### PR TITLE
Add support for PHP 8.2

### PR DESCRIPTION
**Pull request by fredden**
Fixes https://github.com/extdn/github-actions-m2/issues/86

Please let me know if I've missed anything. I don't know if the GitHub Actions will pass straight away, or if someone needs to build and upload the images to Docker Hub first.

### CODE REVIEW

File: examples/extdn-integration-tests-with-elasticsearch.yml [hunk 1]
```diff
@@ -35,3 +35,7 @@ jobs:
         env:
           MAGENTO_VERSION: '2.4.4'
           COMPOSER_VERSION: '2'
+      - uses: extdn/github-actions-m2/magento-integration-tests/8.2@master
+        env:
+          MAGENTO_VERSION: '2.4.6'
```
Comment (sprankhub) [inline examples/extdn-integration-tests-with-elasticsearch.yml:40]:
```suggestion
          MAGENTO_VERSION: '2.4.6-p3'
```

File: examples/extdn-integration-tests-with-marketplace.yml [hunk 2]
```diff
@@ -39,3 +39,7 @@ jobs:
         env:
           MAGENTO_VERSION: '2.4.4'
           COMPOSER_VERSION: '2'
+      - uses: extdn/github-actions-m2/magento-integration-tests/8.2@master
+        env:
+          MAGENTO_VERSION: '2.4.6'
```
Comment (sprankhub) [inline examples/extdn-integration-tests-with-marketplace.yml:44]:
```suggestion
          MAGENTO_VERSION: '2.4.6-p3'
```

File: examples/extdn-integration-tests.yml [hunk 3]
```diff
@@ -29,3 +29,7 @@ jobs:
         env:
           MAGENTO_VERSION: '2.4.4'
           COMPOSER_VERSION: '2'
+      - uses: extdn/github-actions-m2/magento-integration-tests/8.2@master
+        env:
+          MAGENTO_VERSION: '2.4.6'
```
Comment (sprankhub) [inline examples/extdn-integration-tests.yml:34]:
```suggestion
          MAGENTO_VERSION: '2.4.6-p3'
```

File: examples/extdn-unit-tests.yml [hunk 4]
```diff
@@ -20,3 +20,7 @@ jobs:
         env:
           MAGENTO_VERSION: '2.4.4'
           COMPOSER_VERSION: '2'
+      - uses: extdn/github-actions-m2/magento-unit-tests/8.2@master
+        env:
+          MAGENTO_VERSION: '2.4.6'
```
Comment (sprankhub) [inline examples/extdn-unit-tests.yml:25]:
```suggestion
          MAGENTO_VERSION: '2.4.6-p3'
```

File: magento-integration-tests/8.2/action.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  module_name:
+    description: 'Your Magento module name. Example: Foo_Bar'
+    required: true
+  composer_name:
+    description: 'Your composer name. Example: foo/magento2-bar'
+    required: true
+  ce_version:
+    description: 'Magento 2 Open Source version number'
+    required: true
+    default: '2.4.6'
```
Comment (sprankhub) [inline magento-integration-tests/8.2/action.yml:14]:
```suggestion
    default: '2.4.6-p3'
```

File: magento-phpstan/Dockerfile:8.2 [hunk 6]
```diff
@@ -0,0 +1,18 @@
+FROM extdn/magento-integration-tests-action:8.2-latest AS builder
+
+RUN echo memory_limit = -1 >> /usr/local/etc/php/conf.d/custom-memory.ini
+RUN composer2 create-project --repository=https://repo-magento-mirror.fooman.co.nz/ --no-plugins --no-install --no-interaction magento/project-community-edition /var/www/magento2ce "2.4.6"
```
Comment (sprankhub) [inline magento-phpstan/Dockerfile:8.2:4]:
```suggestion
RUN composer2 create-project --repository=https://repo-magento-mirror.fooman.co.nz/ --no-plugins --no-install --no-interaction magento/project-community-edition /var/www/magento2ce "2.4.6-p3"
```

File: magento-unit-tests/8.2/action.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    description: 'Your composer name. Example: foo/magento2-bar'
+    required: true
+  php_version:
+    description: 'PHP version'
+    required: true
+    default: '8.2'
+  magento_version:
+    description: 'Magento 2 Open Source version number'
+    required: true
+    default: '2.4.6'
```
Comment (sprankhub) [inline magento-unit-tests/8.2/action.yml:18]:
```suggestion
    default: '2.4.6-p3'
```

File: magento-integration-tests/Dockerfile:8.2 [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    libjpeg62-turbo-dev \
+    libfreetype6-dev \
+    libzip-dev \
+    libpng-dev \
+    libjpeg-dev \
+    libsodium-dev \
+    libicu-dev \
+    libxml2-dev \
+    libxslt-dev \
+    netcat-openbsd \
```
Comment (sprankhub) [inline magento-integration-tests/Dockerfile:8.2:22]: Just curious: Why not `netcat` any more, but now `netcat-openbsd`?

**Review by sprankhub — CHANGES_REQUESTED** [review]
Thank you so much for picking this up! Looks all good to me, I would just like to default to the newest patch version :)

File: magento-integration-tests/Dockerfile:8.2 (hunk 8 above)
Comment (fredden) [reply magento-integration-tests/Dockerfile:8.2:22]: I started with a clone of the "8.1" file, but got an error when trying to install `netcat`. It seems that there is no package called `netcat`, but there are two virtual packages which provide that name: `netcat-openbsd` and `netcat-traditional`.
https://packages.debian.org/bookworm/netcat

Screenshot_2023-11-06_13-56-54

**Review by sprankhub — APPROVED** [review]

**Pull request merged**